adapter/internal/oasparser/utills/open_api_utils.go

```diff
@@ -51,25 +53,26 @@ func hasPrefix(buf []byte, prefix []byte) bool {
 	return bytes.HasPrefix(trim, prefix)
 }
 
-// FindSwaggerVersion finds the openapi version ("2" or "3") for the given
-// openAPI json content.
-func FindSwaggerVersion(jsn []byte) string {
-	var version string = "2"
+// FindAPIDefinitionVersion finds the API definition version for the given json content.
+func FindAPIDefinitionVersion(jsn []byte) string {
 	var result map[string]interface{}
 
 	err := json.Unmarshal(jsn, &result)
 	if err != nil {
-		logger.LoggerOasparser.Error("json unmarsheliing err when finding the swaggerVersion : ", err)
+		logger.LoggerOasparser.Error("Error while JSON unmarshalling to find the API definition version.", err)
 	}
 
-	if _, ok := result["swagger"]; ok {
-		version = "2"
-	} else if _, ok := result["openapi"]; ok {
-		version = "3"
-	} else {
-		logger.LoggerOasparser.Warn("swagger file version is not defined. Default version set as to 2 ")
-		return version
-
+	if _, ok := result[constants.Swagger]; ok {
+		return constants.Swagger2
+	} else if _, ok := result[constants.OpenAPI]; ok {
+		return constants.OpenAPI3
+	} else if versionNumber, ok := result[constants.AsyncAPI]; ok {
+		if strings.HasPrefix(versionNumber.(string), "2") {
+			return constants.AsyncAPI2
+		}
+		logger.LoggerOasparser.Warn("AsyncAPI version is not supported.")
```

Comment:
Let's print the version as well.

Reply:
And isn't this an error log?

Reply:
Updated to an Error log and printing the version. Fixed with https://github.com/wso2/product-microgateway/pull/2648/commits/a62d44caa9cbfad105440a12fe2851a3d466efa2